src/forms/LocationSearchForm/LocationSearchForm.js

```diff
@@ -1,55 +1,72 @@
 import React from 'react';
-import PropTypes from 'prop-types';
-import { compose } from 'redux';
-import { Field, reduxForm, propTypes as formPropTypes } from 'redux-form';
+import { func, string } from 'prop-types';
+import { Form as FinalForm, Field } from 'react-final-form';
 import { intlShape, injectIntl } from 'react-intl';
 import classNames from 'classnames';
 import { Form, LocationAutocompleteInput } from '../../components';
 
 import css from './LocationSearchForm.css';
 
 const LocationSearchFormComponent = props => {
-  const { rootClassName, className, intl, onSubmit } = props;
-
-  const onChange = location => {
+  const handleChange = location => {
     if (location.selectedPlace) {
       // Note that we use `onSubmit` instead of the conventional
       // `handleSubmit` prop for submitting. We want to autosubmit
       // when a place is selected, and don't require any extra
       // validations for the form.
-      onSubmit({ location });
+      props.onSubmit({ location });
     }
   };
 
-  const classes = classNames(rootClassName || css.root, className);
+  return (
+    <FinalForm
+      {...props}
+      render={({ rootClassName, className, intl, handleSubmit }) => {
```

Comment:
We haven't used this spread format of props with other components. It helps to debug code if there's full prop list available.